tests/geometry/test_intersections.py

```diff
@@ -1,9 +1,2539 @@
+
+        # Intersection is internal to f_1
+        assert len(seg_vert[0][0]) == 0
+        # Intersection on f_2 is on the first vertex
+        assert seg_vert[1][0][0] == 0
+        assert not (seg_vert[1][0][1])
+
+        assert len(seg_vert2[1][0]) == 0
+        assert seg_vert2[0][0][0] == 0
+        assert not (seg_vert2[0][0][1])
+
+    def test_single_point_intersection_on_boundary(self):
+        f_1 = np.array([[0, 0, 0, 0], [0, 1, 1, 0], [0, 0, 1, 1]])
+        f_2 = np.array([[0, 1, 1], [0.5, 0.5, 0.5], [0.0, 0, 1]])
+
+        new_pt, isect_pt, _, _, seg_vert, point_contact = intersections.polygons_3d(
+            [f_1, f_2]
+        )
+        # Also intersect in reverse order
+        (
+            new_pt2,
+            isect_pt,
+            _,
+            _,
+            seg_vert2,
+            point_contact2,
+        ) = intersections.polygons_3d([f_2, f_1])
+        known_point = np.array([[0], [0.5], [0.0]])
+        assert compare_arrays(new_pt, known_point)
+        assert compare_arrays(new_pt2, known_point)
+
+        for arr in point_contact:
+            assert all(arr)
+        for arr in point_contact2:
+            assert all(arr)
+
+        # Intersection is on the first segment of f_1
+        assert seg_vert[0][0][0] == 0
+        assert seg_vert[0][0][1]
+        # Intersection on f_2 is on the first vertex
+        assert seg_vert[1][0][0] == 0
+        assert not (seg_vert[1][0][0])
+
+        assert seg_vert2[1][0][0] == 0
+        assert seg_vert2[1][0][1]
+        assert seg_vert2[0][0][0] == 0
+        assert not (seg_vert2[0][0][1])
+
+    def test_single_point_intersection_on_vertex(self):
+        f_1 = np.array([[0, 0, 0, 0], [0, 1, 1, 0], [0, 0, 1, 1]])
+        f_2 = np.array([[0, 0.5, 1], [0, 0.5, 1], [0, 0, 1]])
+
+        new_pt, isect_pt, _, _, seg_vert, point_contact = intersections.polygons_3d(
+            [f_1, f_2]
+        )
+        # Also intersect in reverse order
+        (
+            new_pt2,
+            isect_pt,
+            _,
+            _,
+            seg_vert2,
+            point_contact2,
+        ) = intersections.polygons_3d([f_2, f_1])
+        known_point = np.array([[0], [0], [0.0]])
+        assert compare_arrays(new_pt, known_point)
+        assert compare_arrays(new_pt2, known_point)
+
+        for arr in point_contact:
+            assert all(arr)
+        for arr in point_contact2:
+            assert all(arr)
+
+        # Intersection is on the first segment of f_1s
+        assert seg_vert[0][0][0] == 0
+        assert not (seg_vert[0][0][1])
+        # Intersection on f_2 is on the first vertex
+        assert seg_vert[1][0][0] == 0
+        assert not (seg_vert[1][0][0])
+
+        assert seg_vert2[0][0][0] == 0
+        assert not (seg_vert2[0][0][1])
+        # Intersection on f_2 is on the first vertex
+        assert seg_vert2[1][0][0] == 0
+        assert not (seg_vert2[1][0][0])
+
+
+class TestPolygonPolyhedronIntersection:
+    def setUp(self):
```

Comment:
Do you have enough of a clue to be able to document a wee bit?

Reply:
No, not really